Strip out carriage returns

teoljungberg opened the pull request:
References #1008

teoljungberg commented on plugin/fugitive.vim:
```diff
@@ -2511,6 +2511,7 @@ function! s:ReplaceCmd(cmd,...) abort
   silent exe 'keepalt file '.tmp
   try
     silent edit!
+    silent exec '%g/'.nr2char(13).'/d'
```
This is a very naive approach, I'd gladly take pointers!

tpope replied:
First pointer is don't delete the entire line.

teoljungberg replied:
🤕 

Fixed in 04ee399 with a normal substitution.

teoljungberg replied:
I also updated it to use the `c` s_flag to only remove the carriage returns if they exist.

teoljungberg replied:
Edit, not `c` but `e`

teoljungberg commented:
Any thoughts on this @tpope? I've been running this branch locally since it was opened without any issues, if you can find any edge-cases for this implementation - or a better implementation - I'm all ears.

tpope commented:
Thinking something like `1,/^diff --git\|\%$/g/\r$/s///`, to avoid munging the diff.

teoljungberg commented:
TIL that you can search for a patterns for ex-commands. Thanks!

I rebased and squashed everything into two commits, one for the unused variable - the other for stripping out the carriage returns.

tpope commented:
Just noticed you're doing this to all sorts of buffers. Try it before https://github.com/tpope/vim-fugitive/blob/3bdac8b60383c0326991ba08bfd57d4e70925f3a/plugin/fugitive.vim#L2732. Based on the context you'll want to add a `keepjumps`.
